primitives/core/src/bounded/bounded_vec.rs

```diff
@@ -276,6 +276,14 @@ impl<'a, T, S> sp_std::iter::IntoIterator for BoundedSlice<'a, T, S> {
 	}
 }
 
+impl<'a, T, S: Get<u32>> BoundedSlice<'a, T, S> {
+	/// Create an instance from the first elements of the given slice (or all of it if it is smaller
+	/// than the length bound).
+	pub fn truncate_from(s: &'a [T]) -> Self {
+		Self(&s[0..(S::get() as usize)], PhantomData)
```

Comment:
This will panic with an out-of-bounds access if the input slice is smaller than the bound.

Implemented a small test do demonstrate the problem:
```
#[test]
fn slice_truncate_from_works() {
	let bounded = BoundedSlice::<u32, ConstU32<4>>::truncate_from(&[1u32, 2u32, 3u32, 4u32, 5u32]);
	assert_eq!(bounded.len(), 4);
	let bounded = BoundedSlice::<u32, ConstU32<4>>::truncate_from(&[1u32, 2u32, 3u32]);
	assert_eq!(bounded.len(), 3);
}
```

This test will currently fail with the following error:
```
---- bounded::bounded_vec::test::slice_truncate_from_works stdout ----
thread 'bounded::bounded_vec::test::slice_truncate_from_works' panicked at 'range end index 4 out of range for slice of length 3', library/core/src/slice/index.rs:73:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Reply:
Will fix. Many thanks!

Reply:
https://github.com/paritytech/substrate/pull/11649/commits/29c56c4e11088eebf83b36dbc5a510429d96485c